File: book_log.py
from datetime import datetime

class BookLog:
    def __init__(self, book, pagesRead=0, totalPagesRead=0, dateAdded=None, completed=False):
        self.id = id
        self.book = book
        self.pagesRead = pagesRead
        self.totalPagesRead = totalPagesRead
        self.dateAdded = dateAdded if dateAdded else datetime.now()
        self.completed = completed

    def calculateTotalPagesRead(self):
        self.totalPagesRead += self.pagesRead
        return self.totalPagesRead

    def triggerInteraction(self):
        

File: BookLog.js
class BookLog {
    constructor(book, pagesRead, timeSpent, dateAdded = new Date(), completed = false) {
        this.book = book;
        this.pagesRead = pagesRead;
        this.timeSpent = timeSpent;
        this.dateAdded = dateAdded;
        this.completed = completed;
    }
}

export default BookLog;